components/percona_telemetry/storage.cc

```diff
@@ -0,0 +1,146 @@
+#include <fstream>
+
+#include "logger.h"
+#include "storage.h"
+
+#include <sys/stat.h>
+#include <ctime>
+#include <filesystem>
+#include <string_view>
+#include <vector>
+
+#include <boost/uuid/random_generator.hpp>
+#include <boost/uuid/uuid.hpp>
+#include <boost/uuid/uuid_io.hpp>
+
+namespace fs = std::filesystem;
+static constexpr auto path_separator =
+    std::filesystem::path::preferred_separator;
+
+// Let's keep it as std::string, because there is no way to string + string_vew
+static const std::string TMP_FILE_EXTENSION(".tmp");
```

Comment:
could be std::string_view

Reply:
Unfortunatley no :(
https://stackoverflow.com/questions/44636549/why-is-there-no-support-for-concatenating-stdstring-and-stdstring-view